### PR TITLE
[UI] Model registry cleanup and tests

### PR DESCRIPTION
**Pull request by aarondav**
## What changes are proposed in this pull request?

Contains a host of model registry bug fixes, cleanup, and unit tests. Additionally, restructures the JavaScript source directory to separate the experiment-tracking and model-registry components more cleanly.

## How is this patch tested?

Lots of unit tests!

### Is this a user-facing change?

- [x] No.

Mainly cleanup and testing additions.

### What component(s) does this PR affect?

- [x] UI

### CODE REVIEW

File: mlflow/server/js/package.json [hunk 1]
```diff
@@ -53,7 +54,7 @@
     "babel-eslint": "8.2.2",
     "enzyme": "3.5.0",
     "enzyme-adapter-react-16": "1.3.0",
-    "eslint": "4.18.2",
+    "eslint": "4.18.1",
```
Comment (Zangr) [inline mlflow/server/js/package.json]:
```suggestion
    "eslint": "4.18.2",
```

File: mlflow/server/js/package.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
     "react-mde": "^7.6.2",
     "react-modal": "^3.4.4",
     "react-overlays": "^0.8.3",
-    "react-plotly.js": "2.4.0",
+    "react-plotly.js": "2.2.0",
```
Comment (Zangr) [inline mlflow/server/js/package.json]:
```suggestion
    "react-plotly.js": "2.4.0",
```

File: mlflow/server/js/src/common/utils/Utils.js [hunk 3]
```diff
@@ -242,6 +244,17 @@ class Utils {
         </a>);
       }
       return res;
+    } else if (sourceType === "JOB") {
```
Comment (Zangr) [inline mlflow/server/js/src/common/utils/Utils.js]: Mark out?

File: mlflow/server/js/src/common/utils/Utils.js [hunk 4]
```diff
@@ -273,13 +288,23 @@ class Utils {
...
+    const sourceType = Utils.getSourceType(tags);
     const entryPointName = Utils.getEntryPointName(tags);
-    if (Utils.getSourceType(tags) === "PROJECT") {
+    if (sourceType === "PROJECT") {
       let res = Utils.dropExtension(Utils.baseName(sourceName));
       if (entryPointName && entryPointName !== "main") {
         res += ":" + entryPointName;
       }
       return res;
+    } else if (sourceType === "JOB") {
```
Comment (Zangr) [inline mlflow/server/js/src/common/utils/Utils.js]: Mark out?

File: mlflow/server/js/src/common/utils/Utils.test.js [hunk 5]
```diff
@@ -158,6 +158,19 @@ test("formatSource & renderSource", () => {
   // Query params must appear before the hash, see https://tools.ietf.org/html/rfc3986#section-4.2
   // and https://stackoverflow.com/a/34772568
   expect(wrapper3.props().href).toEqual("http://localhost/?o=123#notebook/13/revision/42");
+
+  const databricksJobTags = {
```
Comment (Zangr) [inline mlflow/server/js/src/common/utils/Utils.test.js]: Mark out?

File: mlflow/server/js/src/common/utils/TestUtils.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+export const NOOP = () => {};
+
+export const test = NOOP;
+// END-EDGE
```
Comment (Zangr) [inline mlflow/server/js/src/common/utils/TestUtils.js]: What is this used for?  ; )

**Comment by aarondav** [comment]
Closing in favor of #2638

**Pull request closed without merging**